src/Analyzer/Resolve/IdentifierResolveScope.h

```diff
@@ -137,7 +137,7 @@ struct IdentifierResolveScope
     ContextPtr context;
 
     /// Identifier lookup to result
-    std::unordered_map<IdentifierLookup, IdentifierResolveState, IdentifierLookupHash> identifier_lookup_to_resolve_state;
+    std::unordered_map<IdentifierLookup, IdentifierResolveState, IdentifierLookupHash> identifier_in_lookup_process;
```

Comment:
I wonder if we can remove `identifier_in_lookup_process` and `table_expressions_in_resolve_process` from the scope.
Looks like it's part of a single identifier resolution, and when nothing is resolving at the moment, the map is empty.

If so, maybe we can have a separate state for the identifier resolution process to make the scope immutable (or less mutable) and cleaner.

But this is out of scope for this PR